### ISSUE TITLE
TabWidget returns mystery pointer

### ISSUE THREAD

**Issue by svenevs** [wjakob/nanogui#289]
I have no idea how this happens, been staring at it for a while.  It seems that `nullptr` is not always returned [when the tab name "does not exist"](https://github.com/wjakob/nanogui/blob/f62ab889aef5732180b510bc58144d7c229c552c/src/tabwidget.cpp#L96).

---------------------

**Duh**.  Sorry, I've been awake too long.  it seems the fix is obvious

```diff
 const Widget *TabWidget::tab(const std::string &tabName) const {
     int index = mHeader->tabIndex(tabName);
-    if (index == mContent->childCount())
+    if (index == -1 || index == mContent->childCount())
         return nullptr;
     return mContent->children()[index];
 }
```

I'll submit a patch tomorrow, there may be a similar problem in other lookup methods that caused the original problem.

---------------------

I was doing something dubious, where a `ComboBox` attached to a specific tab on a `TabWidget` was trying to perform

1. Delete the existing tab `"Tab Name"`.
2. Recreate the tab `"Tab Name"` at the same index.

This is an odd thing to do, I suppose.  It ended up with some segfaults unless I explicitly `tabWidget->performLayout(ctx)` between 1 and 2.  So I started looking around, but nothing obvious sticks out.  I've engineered a different example that seems to reliably do the wrong thing, perhaps it is the `StackedWidget`?  See `// !!!` in plug-and-play example code below, that tab name has never been created, yet it always prints things out, and the returned tab searches are always `-1`.

```cpp
#include <nanogui/nanogui.h>
#include <iostream>

int main(int argc, const char **argv) {
    nanogui::init();

    {
        // standard setup, create screen / window
        using namespace nanogui;
        Screen *screen = new Screen({600, 600}, "TabWidget Testing");

        Window *window = new Window(screen, "TabWidget Testing");
        window->setFixedSize({400, 400});
        window->setLayout(new GroupLayout());

        // create a tab widget / add a tab with some dummy data
        TabWidget *tabWidget = new TabWidget(window);
        tabWidget->setFixedSize({400 - 30, 400 - 60});
        Widget *layer = tabWidget->createTab("First");
        layer->setLayout(new GroupLayout());
        new Label(layer, "First");
        new Button(layer, "second");
        new Label(layer, "Third");
        new Button(layer, "fourth");

        // !!!
        auto check_it = [tabWidget]() {
            std::string label = "Not Here";
            Widget *should_not_exist = tabWidget->tab(label);
            if (should_not_exist) {
                std::cout << "Tab Label Index: "
                          << tabWidget->tabLabelIndex(label)
                          << std::endl;
                std::cout << "Widget Index:    "
                          << tabWidget->tabIndex(should_not_exist)
                          << std::endl;
            }
        };

        check_it();

        screen->setVisible(true);
        screen->performLayout();

        check_it();

        nanogui::mainloop();
    }

    nanogui::shutdown();
    return 0;
}
```

Basically, it returns to me something that is not possible to find using label searches.  And when I search for the same thing that was returned to me, I also get an index of `-1`.  What's going on here?
